### PR TITLE
Feature/continuousMIIC

### PR DESCRIPTION
**Pull request by MLasserre**
Add the continuous MIIC algorithm to otagrum.
This includes tests, documentation and example.

### CODE REVIEW

File: lib/src/CorrectedMutualInformation.cxx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        //clearCaches();
+        //kmode_ = KModeTypes::Naive;
+    //}
+
+    //void CorrectedMutualInformation::useGaussian() {
+        //clearCaches();
+        //cmode_ = CModeTypes::Gaussian;
+    //}
+
+    //void CorrectedMutualInformation::useBernstein() {
```
Comment (jschueller) [inline lib/src/CorrectedMutualInformation.cxx:42]: could you clean these up ?

File: lib/src/otagrum/ContinuousMIIC.hxx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                                  double, double, double >& e2) const;
+};
+
+class OTAGRUM_API ContinuousMIIC : public OT::Object {
+    public:
+        // Constructor
+        explicit ContinuousMIIC(const OT::Sample &data);
+
+        // Destructor
+        ~ContinuousMIIC();
```
Comment (jschueller) [inline lib/src/otagrum/ContinuousMIIC.hxx:54]: the destructors are useless and should be removed

**Comment by MLasserre** [comment]
It should be good now

**Comment by jschueller** [comment]
some accessor docstrings are missing:
http://openturns.github.io/otagrum/master/user_manual/_generated/otagrum.ContinuousMIIC.html
http://openturns.github.io/otagrum/master/user_manual/_generated/otagrum.CorrectedMutualInformation.html

**Pull request merged**